wallet/chain/x/builder_test.go

```diff
@@ -0,0 +1,583 @@
+				OutputOwners: secp256k1fx.OutputOwners{
+					Locktime:  0,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+					Threshold: 1,
+				},
+			},
+		},
+		{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.Empty.Prefix(utxosOffset + 2),
+				OutputIndex: uint32(utxosOffset + 2),
+			},
+			Asset: avax.Asset{ID: nftAssetID},
+			Out: &nftfx.MintOutput{
+				GroupID: 1,
+				OutputOwners: secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+				},
+			},
+		},
+		{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.Empty.Prefix(utxosOffset + 3),
+				OutputIndex: uint32(utxosOffset + 3),
+			},
+			Asset: avax.Asset{ID: nftAssetID},
+			Out: &secp256k1fx.MintOutput{
+				OutputOwners: secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+				},
+			},
+		},
+		{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.Empty.Prefix(utxosOffset + 4),
+				OutputIndex: uint32(utxosOffset + 4),
+			},
+			Asset: avax.Asset{ID: propertyAssetID},
+			Out: &propertyfx.MintOutput{
+				OutputOwners: secp256k1fx.OutputOwners{
+					Locktime:  0,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+					Threshold: 1,
+				},
+			},
+		},
+		{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.Empty.Prefix(utxosOffset + 5),
+				OutputIndex: uint32(utxosOffset + 5),
+			},
+			Asset: avax.Asset{ID: propertyAssetID},
+			Out: &propertyfx.OwnedOutput{
+				OutputOwners: secp256k1fx.OutputOwners{
+					Locktime:  0,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+					Threshold: 1,
+				},
+			},
+		},
+		{ // a large UTXO last, which should be enough to pay any fee by itself
+			UTXOID: avax.UTXOID{
+				TxID:        ids.Empty.Prefix(utxosOffset + 6),
+				OutputIndex: uint32(utxosOffset + 6),
+			},
+			Asset: avax.Asset{ID: avaxAssetID},
+			Out: &secp256k1fx.TransferOutput{
+				Amt: 9 * units.Avax,
+				OutputOwners: secp256k1fx.OutputOwners{
+					Locktime:  0,
+					Addrs:     []ids.ShortID{utxosKey.PublicKey().Address()},
+					Threshold: 1,
+				},
+			},
+		},
+	}
+}
+
+func newChainUTXOs(require *require.Assertions, utxoSets map[ids.ID][]*avax.UTXO) common.ChainUTXOs {
+	globalUTXOs := common.NewUTXOs()
+	for subnetID, utxos := range utxoSets {
+		for _, utxo := range utxos {
+			require.NoError(
+				globalUTXOs.AddUTXO(stdcontext.Background(), subnetID, constants.PlatformChainID, utxo),
+			)
+		}
+	}
+	return &deterministicChainUTXOs{
+		ChainUTXOs: common.NewChainUTXOs(constants.PlatformChainID, globalUTXOs),
+	}
+}
+
+type deterministicChainUTXOs struct {
+	common.ChainUTXOs
+}
+
+func (c *deterministicChainUTXOs) UTXOs(ctx stdcontext.Context, sourceChainID ids.ID) ([]*avax.UTXO, error) {
+	utxos, err := c.ChainUTXOs.UTXOs(ctx, sourceChainID)
+	if err != nil {
+		return nil, err
+	}
+
+	slices.SortFunc(utxos, func(a, b *avax.UTXO) int {
+		return a.Compare(&b.UTXOID)
+	})
+	return utxos, nil
+}
```

Comment:
Should we move these into the `common` package (or perhaps a testutil package for the wallet) rather than duplicating them?